packages/dds/tree/src/test/shared-tree/sharedTree.spec.ts

```diff
@@ -1090,6 +1094,216 @@ describe("SharedTree", () => {
+					const content = {
+						schema,
+						allowedSchemaModifications: AllowedUpdateType.None,
+						initialTree: [["a"]],
+					};
+					const resubmitTree = provider.trees[0].schematizeInternal(content);
+					const otherTree = provider.trees[1].schematizeInternal(content);
+					provider.processMessages();
+
+					// validate insertion
+					validateTreeContent(otherTree.checkout, content);
+
+					const otherUndos = makeEdits(otherTree, otherPeer);
+					const resubmitUndos = makeEdits(resubmitTree, resubmitPeer);
+
+					provider.processMessages();
+
+					if (otherPeer === "A" && resubmitPeer === "A") {
+						assert.deepEqual(
+							// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+							[...otherTree.flexTree.content.content.at(0)!.content],
+							["a", "b", "c", "b", "c"],
+						);
+						assert.deepEqual(
+							// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+							[...resubmitTree.flexTree.content.content.at(0)!.content],
+							["a", "b", "c", "b", "c"],
+						);
+					} else {
+						assert.deepEqual([...resubmitTree.flexTree.content.content], []);
+						assert.deepEqual([...otherTree.flexTree.content.content], []);
+					}
+
+					assert.equal(otherUndos[1].revert(), RevertibleResult.Success);
+					assert.equal(otherUndos[0].revert(), RevertibleResult.Success);
+					assert.equal(
+						resubmitUndos[resubmitPeer === "B" ? 1 : 0].revert(),
+						RevertibleResult.Success,
+					);
+					provider.processMessages();
+
+					// disconnect the resubmit tree
+					provider.trees[0].setConnected(false);
+
+					assert.equal(
+						resubmitUndos[resubmitPeer === "B" ? 0 : 1].revert(),
+						RevertibleResult.Success,
+					);
+					const otherTreeRoot = otherTree.flexTree.content.content.at(0);
+					if (otherTreeRoot !== undefined) {
+						assert.notDeepEqual([...otherTreeRoot.content], ["a"]);
+					}
+
+					provider.trees[0].setConnected(true);
+					provider.processMessages();
+
+					assert.deepEqual(
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						[...resubmitTree.flexTree.content.content.at(0)!.content],
+						["a"],
+					);
+					assert.deepEqual(
+						// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+						[...otherTree.flexTree.content.content.at(0)!.content],
+						["a"],
+					);
+				});
+			});
+
+			it("the restore of a tree edited on a branch", () => {
+				const provider = new TestTreeProviderLite(2);
+				const content = {
+					schema,
+					allowedSchemaModifications: AllowedUpdateType.None,
+					initialTree: [["a"]],
+				};
+				const resubmitTree = provider.trees[0].schematizeInternal(content);
+				const { undoStack, unsubscribe } = createTestUndoRedoStacks(
+					resubmitTree.checkout.events,
+				);
+				const otherTree = provider.trees[1].schematizeInternal(content);
+				provider.processMessages();
+
+				// validate insertion
+				validateTreeContent(otherTree.checkout, content);
```

Comment:
Same as in the previous test: remove unless this is material to the test.
Just to be clear, I'm not against asserting the state from time to time even if the functionality under test should not influence that test. I think that's a good thing to do when some edits have gove by and the reader of the test might wellcome some auto-validated documentation about what the current state ought to be, but here it's pretty obvious what the state ought to be.